src/AutorestSwift/View Models/ServiceClientViewModel.swift

```diff
@@ -42,23 +46,184 @@ struct ParameterViewModel {
+    // if match, the value is taken from the signature parameter
+    for param in params where param.name == schema.name {
+        return param.name
+    }
+
+    return ""
+}
+
+struct RequestViewModel {
+    let path: String
+    let method: String
+    let knownMediaType: String?
+    let uri: String?
+    let mediaTypes: [String]?
+    let params: [ParameterViewModel]?
+    let objectType: String?
+
+    init(from request: Request) {
+        // load HttpRequest properties
+        if let httpRequest = (request.protocol.http as? HttpRequest) {
+            self.path = httpRequest.path
+            self.method = httpRequest.method.rawValue
+            self.uri = httpRequest.uri
+        } else {
+            self.path = ""
+            self.method = ""
+            self.uri = ""
+        }
+
+        // load HttpWithBodyRequest specfic properties
+        if let httpWithBodyRequest = (request.protocol.http as? HttpWithBodyRequest) {
+            self.mediaTypes = httpWithBodyRequest.mediaTypes
+            self.knownMediaType = httpWithBodyRequest.knownMediaType.rawValue
+        } else {
+            self.mediaTypes = nil
+            self.knownMediaType = nil
+        }
+
+        // check if the request body is from signature parameter. If yes, store the object type
+        // and add the request signature parameter to operation parameters
+        var params = [ParameterViewModel]()
+        for param in request.signatureParameters ?? [] {
+            params.append(ParameterViewModel(from: param))
+        }
+        self.params = params
+
+        if request.signatureParameters?.count == 1,
+            request.signatureParameters?[0].schema.type == AllSchemaTypes.object {
+            self.objectType = request.signatureParameters?[0].schema.name
+        } else {
+            self.objectType = nil
+        }
+    }
+}
+
+struct ResponseViewModel {
+    let statusCodes: [String]
+    let knownMediaType: String?
+    let mediaTypes: [String]?
+    let objectType: String?
+
+    init(from response: Response) {
+        if let httpResponse = (response.protocol.http as? HttpResponse) {
+            var statusCodes = [String]()
+            for code in httpResponse.statusCodes {
+                statusCodes.append(code.rawValue)
+            }
+            self.statusCodes = statusCodes
+            self.knownMediaType = httpResponse.knownMediaType?.rawValue
+            self.mediaTypes = httpResponse.mediaTypes
+        } else {
+            self.statusCodes = []
+            self.knownMediaType = nil
+            self.mediaTypes = nil
+        }
+
+        // check if the request body schema type is object, store the object type of the response body
+        if let schemaResponse = response as? SchemaResponse,
+            schemaResponse.schema.type == AllSchemaTypes.object {
+            self.objectType = schemaResponse.schema.name
+        } else {
+            self.objectType = nil
+        }
```

Comment:
Same here

Reply:
This one should still be refactored like the other instances.

Reply:
sorry for the miss fix.